src/components/Auth/index.js

```diff
@@ -0,0 +1,161 @@
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+
+// const AuthService = require('./service');
+const AuthHelper = require('./helper');
+const UserService = require('../User/service');
+const AuthValidation = require('./validation');
+const AuthUtils = require('./utils');
+const ValidationError = require('../../error/ValidationError');
+const { tokens } = require('../../config/token');
+
+async function signUp(req, res, next) {
+    try {
+        const { error, value } = AuthValidation.signUp(req.body);
+
+        if (error) {
+            throw new ValidationError(error.details);
+        }
+
+        const salt = await bcrypt.genSalt(10);
+        const hashedPassword = await bcrypt.hash(req.body.password, salt);
```

Comment:
try use mongoose hooks for generate password, pre-save 
https://mongoosejs.com/docs/middleware.html#defining